lesson_13/adrian_filimon/lesson_13.py

```diff
@@ -0,0 +1,65 @@
+from math import sqrt
+
+class PerimeterMixin:
+    def perimeter(self):
+
+        class_name = type(self).__name__
+
+        return f"This is the perimeter for {class_name}: {sum(self.sides)}"
+
+
+class Polygon(PerimeterMixin):
+    def __init__(self, *args):
+        self.sides = args
+
+    def __str__(self):
+        no_of_sides = len(self.sides)
+        return f"{no_of_sides} is the number of sides."
+
+    def display(self):
+        for side_index, length in enumerate(self.sides, start=1):
+            print(f"Side {side_index} with length: {length}")
+
+
+class Triangle(Polygon):
+    # `s(s-a)(s-b)(s-c) ** 0.5` where s = `(a+b+c) / 2`
+    def __init__(self, *args):
+        super().__init__(*args)
+
+    def area(self):
+        a, b, c = self.sides
+        s = sum(self.sides) / 2
+        return s * (s - a) * (s - b) * (s - c) ** 0.5
+
+
+class Square(Polygon):
+    def __init__(self, *args):
+        super().__init__(*args)
+
+    def area(self):
+        side, *_ = self.sides
+        return side ** 2
+
+    @classmethod
+    def from_area(cls, area):
+        sq = sqrt(area)
+        return f"Length of side is {sq}"
```

Comment:
You added the class method but the return is still a string. it is better to return an instance. `cls(sq,sq,sq,sq)`

Reply:
@AdrianFilimon1707 pinging you just as a reminder to take a look at this PR :smile: :smile: :smile: